pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -389,6 +401,8 @@ def dump_checkpoint(self, weights_only: bool = False) -> dict:
         model.on_save_checkpoint(checkpoint)
         if self.trainer.datamodule is not None:
             self.trainer.datamodule.on_save_checkpoint(checkpoint)
+        if not weights_only:
+            self.trainer.precision_plugin.on_save_checkpoint(checkpoint)
```

Comment:
why does this happen after the other on_save_checkpoint hooks? let's keep the same order as before